src/containerapp/azext_containerapp/_up_utils.py

```diff
@@ -707,6 +707,8 @@ def _get_app_env_and_group(
             matched_apps = [c for c in matched_apps if (c.get("extendedLocation") and c["extendedLocation"]["name"].lower() == env.custom_location_id.lower())]
         elif env.resource_type:
             matched_apps = [c for c in matched_apps if parse_resource_id(c["properties"]["environmentId"])["resource_type"].lower() == env.resource_type.lower()]
+        if custom_location.connected_cluster_id:
+            matched_apps = _filter_containerapps_by_connected_cluster_id(cmd, matched_apps, custom_location.connected_cluster_id)
```

Comment:
is it possible to have two custom location in different connected clusters?

Reply:
Yes. If user input custom-location, we filter by custom location first.